libnavigation-base/src/main/java/com/mapbox/navigation/base/trip/model/roadobject/UpcomingRoadObject.kt

```diff
@@ -1,13 +1,21 @@
 package com.mapbox.navigation.base.trip.model.roadobject
 
 /**
- * Holds the road objects and the distance to the point where the alert occurs,
- * or start an alert if it has length.
+ * Holds the road objects and the distance to the start of the object.
+ * There are two sources of road objects: active route and the electronic horizon.
+ * Road objects coming from the electronic horizon will also have [RoadObjectDistanceInfo]
+ * Available [RoadObjectDistanceInfo] types are:
+ * - [RoadObjectDistanceInfoType.GANTRY]
+ * - [RoadObjectDistanceInfoType.LINE]
+ * - [RoadObjectDistanceInfoType.POINT]
+ * - [RoadObjectDistanceInfoType.POLYGON]
+ * - [RoadObjectDistanceInfoType.SUB_GRAPH]
  *
  * @param roadObject road object
  * @param distanceToStart remaining distance to the start of the object.
  * @param distanceInfo provides extra distance details for road objects coming from
- * Electronic Horizon.
+ * Electronic Horizon. Will be null for objects that are on the current route that we are actively
```

Comment:
Does that mean that e-horizon will not return those objects? I don't think so, right?

Reply:
no, 
`distanceInfo`
will always be null for directions road objects
and non-null for EH

Reply:
doc updated